.github/workflows/ci.yml

```diff
@@ -56,12 +56,23 @@ jobs:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 
 
     - name: Dependencies [Ubuntu]
-      if: matrix.os == 'ubuntu-22.04'
+      if: matrix.os == 'ubuntu-24.04'
       shell: bash
       run: |
         sudo apt update
         sudo apt install -y git build-essential cmake libace-dev coinor-libipopt-dev libeigen3-dev libopencv-dev qtbase5-dev \
-                            qtdeclarative5-dev qtmultimedia5-dev libtinyxml-dev libgsl-dev libpython3-dev swig
+                            qtdeclarative5-dev qtmultimedia5-dev libtinyxml-dev libgsl-dev libpython3-dev
+
+    - name: Install Custom SWIG 4.2.1 [Ubuntu]
+      if: matrix.os == 'ubuntu-24.04'
+      shell: bash
+      run: |
+        mkdir -p ${GITHUB_WORKSPACE}/swig
+        cd ${GITHUB_WORKSPACE}/swig
+        wget https://github.com/robotology/robotology-vcpkg-ports/releases/download/storage/swig_4_2_1_ubuntu_24_04.zip
+        unzip swig_4_2_1_ubuntu_24_04.zip
+        export PATH=${GITHUB_WORKSPACE}/swig/bin:$PATH
+        export SWIG_LIB=${GITHUB_WORKSPACE}/swig/share/swig/4.2.1
```

Comment:
Hi @traversaro 

The CI is still failing.

Do you if I have to set up the CMake params to build YARP as done in https://github.com/robotology/robotology-superbuild/blob/master/cmake/RobotologySuperbuildLogic.cmake#L60-L61 or setting the env vars `PATH` and `SWIG_LIB` as done here should be sufficient?

Reply:
I realized now that the zip contains a further level of path.

Reply:
We've got still v4.2.0 installed somewhere.

![image](https://github.com/user-attachments/assets/e7495bb2-f263-4779-af7f-5b7c4b2dbda5)

Reply:
It's in the runner image:
- https://github.com/actions/runner-images/issues/1203

Reply:
The `export` command is not persistent across jobs steps. To set the `PATH` you either need to do  this in the step in which you actually use SWIG (like in the cmake step) or by setting the PATH persistently via https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#adding-a-system-path and https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#example-of-writing-an-environment-variable-to-github_env .

Reply:
Correct! I'll do it!